components/gallery/GalleryForm.tsx

```diff
@@ -132,6 +152,49 @@ function GalleryForm(props: GalleryFormProps) {
 
     setIsUploading(false);
   }
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  async function uploadNFT(selectedNFT: OwnedNft) {
+    setIsUploadingNFT(true);
+    handleCloseModal();
+    try {
+      if (!selectedNFT?.media?.[0]?.raw) {
+        throw Error("No Media attached to asset");
+      }
+      const raw = selectedNFT?.media?.[0]?.raw.slice(6);
+      const url = process.env.NEXT_PUBLIC_IPFS_GATEWAY + "/ipfs" + raw;
+
+      const res = await axios.get(url, { responseType: "arraybuffer" });
+      const data = res.data;
+      const blob = new Blob([data]);
+
+      const type = res.headers["content-type"];
+      const fileExt = type.split("/")[1];
+
+      const file = new File([blob], `${selectedNFT.title}.${fileExt}`, {
+        type: type,
+      });
+
+      const format = getFormat(file.name);
+      const { encoding } = format;
+      setFileFormat(encoding ?? null);
+      // Upload to Web3 storage
+      const added = await uploadFile(w3InvocationConfig, file);
```

Comment:
As mentioned, this is no longer needed. The `selectedNFT?.media?.[0]?.raw` value can be passed directly to `content` below.

Reply:
Hey Cody, thanks for the guidance. I am testing this method now and am getting a CID parsing error when I click the Done button to add the asset. 
![image](https://github.com/Geo-Web-Project/cadastre/assets/32372824/4c41ce35-cf26-42a3-9a2c-4533a05723e3)

Here is the media object of the test NFT I am using. How do you recommend dealing with a case like this that has the file path included in the cid? Should I remove both the ipfs:// and the filename if present and pass that directly to content?

{
    "gateway": "https://nft-cdn.alchemy.com/opt-goerli/36bba54ec895793b9993062ca2ca6a7e",
    "thumbnail": "https://res.cloudinary.com/alchemyapi/image/upload/thumbnailv2/opt-goerli/36bba54ec895793b9993062ca2ca6a7e",
    "raw": "ipfs://QmVdntsdbBPjrXoHuLSxk2Jr6HwZZD7MoJRguRXDeGU4sb/punks.webp",
    "format": "webp",
    "bytes": 266
}

Reply:
Looks like the `newMediaGallery` prop is using the `MediaObject` type which is expecting a `CID` type for the `content` field. You could try just removing/replacing this type with one that expects a string instead, and remove the `CID.parse`.

Reply:
The filename should be fine and can be kept. As long as the string is a URI that begins with a protocol scheme.